### PR TITLE
Fix Python SDK

### PR DESCRIPTION
**Pull request by davidiw**
* Found bugs in the last update to Python and fixed those
* Finished up the process of bringing in mypy and flushing out good type checking

Tested with
`make test && make examples`

I'm not competent enough to figure out how to integrate with CI/CD

### CODE REVIEW

File: .github/workflows/check-sdk-examples.yaml [hunk 1]
```diff
@@ -49,3 +44,22 @@ jobs:
           max_attempts: 5
           timeout_minutes: 20
           command: cd ./ecosystem/typescript/sdk/examples/javascript && pnpm install && pnpm test
+
+  run-python-examples:
+    runs-on: ubuntu-latest
+    env:
+      APTOS_NODE_URL: https://fullnode.devnet.aptoslabs.com
```
Comment (banool) [inline .github/workflows/check-sdk-examples.yaml:51]: I didn't realize the Python SDK was smart enough to add the `/v1`?

File: ecosystem/python/sdk/aptos_sdk/client.py [hunk 2]
```diff
@@ -250,7 +252,7 @@ def transfer(self, sender: Account, recipient: AccountAddress, amount: int) -> s
         }
         return self.submit_transaction(sender, payload)
 
-    #:!:>bcs_transfer
+    # :!:>bcs_transfer
```
Comment (banool) [inline ecosystem/python/sdk/aptos_sdk/client.py:255]: I think `black` does this but it breaks the templating into the tutorials, have we double checked that? I _think_ that's why I wasn't using `black` on the `aptos_sdk` files in the first place.

Comment (davidiw) [reply ecosystem/python/sdk/aptos_sdk/client.py:255]: it worked :)

File: .github/workflows/check-sdk-examples.yaml (hunk 1 above)
Comment (davidiw) [reply .github/workflows/check-sdk-examples.yaml:51]: I think you added that...

Comment (banool) [reply .github/workflows/check-sdk-examples.yaml:51]: lol okay go me

**Review by banool — APPROVED** [review]

**Pull request merged**